### PR TITLE
Rewrite libtrakt and introduced generic recommended shows tempate

### PR DESCRIPTION
**Pull request by p0psicles**
Rewrite libtrakt to remove all the sickbeard settings from the lib, and introduced generic recommended shows tempate

Goal of this commit is to rewrite the libtrakt because it had sickbeard global settings in it, and other imports, that shouldnt be part of a lib.

At the same time work has been done on the recommendations framework, in /sickrage/show/recommendations code.

Principles where:
* No new functionality should be introduced.
* The libtrakt lib should really be used like a lib. Meaning for ex. no sickbeard.TRAKT_ACCESS_TOKEN in the lib.
* The template for trakt, should be able to be reused for other sources like anidb recommended shows or imdb watchlists.

changes:
* Introduced a generic template. This one should look the same as addShows_recommended.mako
* Lot's of PEP8 changes
* notifier changes to adapt to the libtrakt changes
* traktChecker changes, to adapt to the libtrakt changes. Altough 99% of that isn't used anymore, because traktScheduler has been disabled.
* changes to core.js, to make sure shows can be added from trakt, through the addShowByID webserve function. That function has been changed in webserve, to accept default show options params.

TODO:

- [ ]  Fix some of the libtrakt exception docstrings. Some are to detailed, or just plain wrong.
- [ ]  test
- [ ]  Change double to single quotes

### CODE REVIEW

**Comment by p0psicles** [comment]
Would appreciate some feedback? @labrys ?

**Comment by fernandog** [comment]
https://github.com/pymedusa/SickRage/pull/135

**Comment by p0psicles** [comment]
Made sure traktChecker() now also properly uses the modified lib. I haven't paid much attention to this, as other forks didn't use the trakt syncing anymore. Also fixed a bug in the process. https://github.com/pymedusa/SickRage/pull/378

File: sickrage/show/recommendations/recommended.py [hunk 1]
```diff
@@ -34,8 +34,8 @@ class RecommendedShow(object):
     """
     Base class for show recommendations
     """
-    def __init__(self, show_id, title, indexer, indexer_id, cache_subfolder=u'recommended',
-                 rating=None, votes=None, image_href=None, image_src=None):
+    def __init__(self, rec_show_prov, show_id, title, indexer, indexer_id, cache_subfolder=u'recommended',
```
Comment (p0psicles) [inline sickrage/show/recommendations/recommended.py]: need to change this to kwargs

File: sickrage/show/recommendations/trakt.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+                except MultipleShowObjectsException:
+                    continue
+
+            blacklist = sickbeard.TRAKT_BLACKLIST_NAME not in ''
+
+        except TraktException as e:
+            logger.log(u"Could not connect to Trakt service: %s" % ex(e), logger.WARNING)
+
+        return (blacklist, trending_shows)
```
Comment (p0psicles) [inline sickrage/show/recommendations/trakt.py]: Need to change this to named_tuple

**Pull request closed without merging**